test/integration/integration_test.cc

```diff
@@ -24,21 +28,21 @@ TEST_F(IntegrationTest, Echo) {
   EXPECT_EQ("hello", response);
 }
 
-TEST_F(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
+TEST_P(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
```

Comment:
Ah OK, I see, you only parameterized the echo listener, so that even though all the tests are running through v4 and v6, the other ones use a v4 loopback for the other listeners.

If we are going to be explicit, and now that we are doing full integration tests, I would recommend that you take this opportunity to set IPV6_V6ONLY on v6 sockets. This will prevent mapped v4 addresses from still working. It will double check what you are doing and make all the tests better. Later, we can potentially add a listener option to allow dual bind, but for now I agree we should be explicit.